source/ways-of-working/troubleshooting/index.html.md.erb

```diff
@@ -0,0 +1,97 @@
+   - Look at flux logs to see if there are any errors specific to your release  
+      
+     ```shell
+     kubectl logs -n admin -l app=flux --tail=100 | grep <Release Name>
+     ```
+       
+   - If you see an error which says `failed to resolve reference` or `not found` on a particular image, it means the current image deployed is deleted from the container registry and you need to manually update to latest image in cnp-flux-config to fix your helm release.
+   - If you don't see any error logs for your specific release, check if there are any errors that could be blocking flux.
+       
+     ```shell
+     kubectl logs -n admin -l app=flux --tail=100
+     ```
+        
+   - If there are no errors, it is likely that flux pod is stuck and a symptom for it is not seeing a log containing `cmd="kubectl apply -f -` in flux logs. You should [ask for help from platops](../asking-for-help)) to get it sorted.
+      
+      ```shell
+      kubectl logs -n admin -l app=flux --tail=100
+      ```
+  
+### Updated HelmRelease is not deployed
+    
+   - Helm operator queues all the updates, so it could take up to 20 minutes sometimes to be picked up.
+   - Look at helm operator logs to see if there are any errors specific to your helm release
+   
+      ```shell
+      kubectl logs -n admin -l app=helm-operator --tail=1000 | grep <Release Name>
+      ```
+   - If you see any errors like, `status 'pending-install' of release does not allow a safe upgrade"`. You need to delete `HelmRelease` for fixing this, request platops for help if you do not have permissions.
+     
+      ```shell
+      kubectl delete hr <helm-release-name> -n <namespace>
+      ```
+   - In most cases, helm release gets timedout with an error in log similar to ` failed: timed out waiting for the condition`. This usually means application pods didn't startup in time and you need to look at your pods to know more.
+  
+     <b>Below are some commands to help you debug the issue :</b>
+     
+     Check the latest status on helm release and if it has already been rolled back to previous release.
+     
+     ```shell
+     kubectl describe hr <helm-release-name> -n <your-namespace>
+     ```
+     
+     To check latest events on your namespace
+     
+     ```shell
+     kubectl get events -n <your-namespace>
+     ```
+     
+     To check status of pods
+     
+     ```shell
+     kubectl get pods -n <your-namespace> | grep <helm-release-name>
+     ```
+     
+     To check status of a specific pod which is not running 
+     
+     ```shell
+     kubectl describe pod <pod-name> -n <your-namespace>
+     ```
+   - If you are looking at pods after a long time, `HelmRelease` might have been rolled back and you won't have failed pods. Easiest way is to add a simple change like a dummy environment variable in flux-config to re-trigger the release and debug the issue when it occurs.
```

Comment:
Should we look at adding in helm history cmd to confirm if helm has rolled back?

Reply:
it will be in the helm release status as well like. Just avoiding `helm` commands as they might not enough privileges for `helm` commands.  

![image](https://user-images.githubusercontent.com/47391951/112479025-83135a00-8d6c-11eb-8fa9-43ce144cf42d.png)